store/src/java/com/zimbra/cs/service/mail/ModifyProfileImage.java

```diff
@@ -118,10 +129,18 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                 prefs.put(Provisioning.A_thumbnailPhoto, null);
                 Provisioning.getInstance().modifyAttrs(mbox.getAccount(), prefs, true,
                     zsc.getAuthToken());
+            } else if (removeFromLDAPSuccess && !removeFromDBSuccess) {
+                // Database removal failed; revert to LDAP
+                HashMap<String, Object> prefs = new HashMap<String, Object>();
+                prefs.put(Provisioning.A_thumbnailPhoto, ldapBackup);
```

Comment:
Do we really need to store the default thumbnail in the prefs for every account?  Seems to me, the absense of it should imply "use the default".

Reply:
this code is basically restoring the thumbnail image to LDAP in case of database image removal fails. To make sure that if the image is deleted, it is deleted from LDAP as well as database, otherwise not deleted from anywhere.